activestorage/app/models/active_storage/blob.rb

```diff
@@ -246,7 +246,7 @@ def upload(io, identify: true)
   end
 
   def unfurl(io, identify: true) # :nodoc:
-    self.checksum     = compute_checksum_in_chunks(io)
+    self.checksum     = service&.compute_checksum_in_chunks(io)
```

Comment:
Similar to https://github.com/rails/rails/pull/54468/files#r1947500486, I wonder how we might improve the `compute_checksum_in_chunks` naming. 

Since it was previously a `private` method, it's likely that its name served as its only documentation. If it were promoted to the public API, it could be more thoroughly documented and wouldn't need to communicate argument type expectations in its name.

What do you think about omitting the `_in_chunks` suffix?

```suggestion
    self.checksum     = service&.compute_checksum(io)
```

Reply:
Would it be clear what the difference would be between `base64digest` vs `compute_checksum`?

We could keep `compute_checksum_in_chunks` private on the base service class and expand `base64digest` to something like
`def base64digest(io, in_chunks: false)`

Reply:
I think it would be simpler if we assume `checksum_implementation` respond to `new` and `<<`.

Reply:
I looking to avoid having to call `.base64digest` on the return of `file` or `<<` (in compute_checksum_in_chunks/compute_checksum method) but it would be possible to remove `base64digest` and `file` public methods on the services if we wanted to change calls from:
`service.file(file)` to `service.checksum_implementation.file(file).base64digest)`
and
`service.base64digest(io)` to `service.checksum_implementation.base64digest(io)`